common/python/configs.py

```diff
@@ -241,6 +264,10 @@ def __init__(self, name, number, type, description, extra_config):
         #: Store the extension register info
         self.extension = extra_config.pop("extension", None)
         self.extension_reg = extra_config.pop("extension_reg", None)
```

Comment:
is `extension_reg` actually useful? or is it just a relic from old code? I see it is used in `fmc_acq427` but I don't see any other code needing it, so we might get rid of it

Reply:
I had a deeper look and I think we can get read of all appearances of `extension_reg`... it is redundant because when we specify `extension` we are already telling that it's not a real hardware register... we can see that the `system` block doesn't use it and it works fine

Reply:
I'm going to have a look at this, I remember previously having a look and coming to the conclusion that extension_reg was necessary. But I can't remember why now. If it isn't needed I'll remove it, if it is needed I will add some documentation on what it is doing....

Reply:
I believe `extension_reg` is used when the field is also a hardware register. This is described in the [documentation](https://pandablocks.github.io/PandABlocks-FPGA/master/reference/block.html#field-sections) - although it looks like I've removed the line as part of this pull request. The `fmc_acq427` block needs hardware registers for some of its extension fields.  The `system` block doesn't need hardware registers so doesn't use `extension_reg`. I think this is still the case and `extension_reg` is still needed?

Reply:
Just wanted to reiterate what we talked about earlier:
- yes, `extension_reg` creates a harware register, but the server doesn't know about it and doesn't use it
- in the specific case of `fmc_acq427`, those created registers are used for right shifting the output (i.e. multiplying by a power of 2), however, because it is never set by the server, it uses the default value of 0 and shifting by 0 leaves the same value intact
- I don't understand why we would want to do that right shifting (considering the source data is 32 bits), but if we really need that feature, we could easily do that inside the extension

Reply:
In summary, the old `extension_reg` is deprecated by the type `write_extension` and the extension returns the value to write